labrad/protocol.py

```diff
@@ -424,19 +518,25 @@ def do_connect():
     @inlineCallbacks
     def start_tls(p, cert_string=None):
         try:
-            resp = yield p.sendRequest(C.MANAGER_ID, [(1L, ('STARTTLS', host))])
+            cert = yield p._sendManagerRequest(1, ('STARTTLS', host))
         except Exception:
             raise Exception(
                 'Failed sending STARTTLS command to server. You should update '
                 'the manager and configure it to support encryption or else '
                 'disable encryption for clients. See '
                 'https://github.com/labrad/pylabrad/blob/master/CONFIG.md')
-        cert = resp[0][1]
         p.transport.startTLS(crypto.tls_options(host, cert_string=cert_string))
         returnValue(cert)
 
+    @inlineCallbacks
     def ping(p):
-        return p.sendRequest(C.MANAGER_ID, [(2L, 'PING')])
+        resp = yield p._sendManagerRequest(2, 'PING')
+        features = resp.partition(':')[-1]
```

Comment:
Ok, changed the ping response to look like `("PONG", ["feature1", "feature2"])`. It's easy to distinguish this from the response of old managers. I agree it's better than require clients to parse a string.